### PR TITLE
Fix crash when add to many files

### LINKED ISSUE #4: When add too many files program crashes
Just copy many files and throw them to the column to make the program crash.

### PR DESCRIPTION
**Pull request by Ra-V2**
Fixes #4

### CODE REVIEW

**Comment by owenca** [comment]
You should find the cause of the crash and fix it there.

**Comment by owenca** [comment]
This is definitely not the direction to go. Imagine if we had to add a try-catch block for every `new` operator we used...

I'd figure out how the code is handling dropping multiple files on the menu bar. For example, for your previous attempt of limiting the number of items to no more than 33, dropping 32 files at the top of an empty menu bar followed by dropping 33 different files at the bottom (the 33rd button) would still cause the program to crash in the same way.

File: src/ItemView.cpp [hunk 1]
```diff
@@ -339,7 +339,10 @@ void CItemView::SetData( BPath &p )
 	if( !bIsEmpty )
 	{
```
Comment (owenca) [inline src/ItemView.cpp:340]:
```suggestion
	{
		if (mDockPos >= SLOT_COUNT - 1)
			return;
```

File: src/ItemView.cpp [hunk 2]
```diff
@@ -339,7 +339,10 @@ void CItemView::SetData( BPath &p )
...
 		BPath path = mPath.Path();
-		( ( CItemView *)mItemListView->mItemList.ItemAt( ( mDockPos+1 ) % SLOT_COUNT ) )->SetData( path );
+		if (mDockPos < SLOT_COUNT-1)
+			( ( CItemView *)mItemListView->mItemList.ItemAt( ( mDockPos+1 ) % SLOT_COUNT ) )->SetData( path );
+		else
+			return;
```
Comment (owenca) [inline src/ItemView.cpp]:
```
		((CItemView*) mItemListView->mItemList.ItemAt(mDockPos + 1))->SetData(path);
```

**Review by owenca — CHANGES_REQUESTED** [review]
I'm OK with this, but please see the requested changes below.

**Comment by owenca** [comment]
Please review my suggestions. Basically,
```
if (…)
    return;
BPath path = …;
((CItemView*) …)->…;
```

**Review by owenca — APPROVED** [review]

**Comment by humdingerb** [comment]
Thanks, guys!

**Pull request merged**